algorithms/bfs/count_islands.py

```diff
@@ -0,0 +1,68 @@
+"""
+This is a bfs-version of counting-islands problem in dfs section.
+Given a 2d grid map of '1's (land) and '0's (water),
+count the number of islands.
+An island is surrounded by water and is formed by
+connecting adjacent lands horizontally or vertically.
+You may assume all four edges of the grid are all surrounded by water.
+
+Example 1:
+11110
+11010
+11000
+00000
+Answer: 1
+
+Example 2:
+11000
+11000
+00100
+00011
+Answer: 3
+
+Example 3:
+111000
+110000
+100001
+001101
+001100
+Answer: 3
+
+Example 4:
+110011
+001100
+000001
+111100
+Answer: 5
+"""
+
+
+def counting_islands(grid):
```

Comment:
change the method name to `count_islands`

Reply:
Thanks for your comment. I'll revise and reload it soon :)